internal/backend/connections/connections.go

```diff
@@ -44,6 +60,57 @@ func (c *Connections) Delete(ctx context.Context, id sdktypes.ConnectionID) erro
 	return c.DB.DeleteConnection(ctx, id)
 }
 
+func (c *Connections) List(ctx context.Context, filter sdkservices.ListConnectionsFilter) ([]sdktypes.Connection, error) {
+	conns, err := c.DB.ListConnections(ctx, filter, false)
+	if err != nil {
+		return nil, err
+	}
+
+	return kittehs.TransformError(conns, func(conn sdktypes.Connection) (sdktypes.Connection, error) {
+		return c.enrichConnection(ctx, conn)
+	})
+}
+
+func (c *Connections) getIntegration(ctx context.Context, id sdktypes.ConnectionID) (sdkservices.Integration, error) {
+	conn, err := c.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.Integrations.GetByID(ctx, conn.IntegrationID())
+}
+
+func (c *Connections) Test(ctx context.Context, id sdktypes.ConnectionID) (sdktypes.Status, error) {
+	i, err := c.getIntegration(ctx, id)
+	if err != nil {
+		return sdktypes.InvalidStatus, err
+	}
+
+	if i.Get().ConnectionCapabilities().SupportsConnectionTest() {
+		return i.TestConnection(ctx, id)
+	}
+
+	return sdktypes.InvalidStatus, sdkerrors.ErrNotImplemented
+}
+
+func (c *Connections) RefreshStatus(ctx context.Context, id sdktypes.ConnectionID) (sdktypes.Status, error) {
+	i, err := c.getIntegration(ctx, id)
+	if err != nil {
+		return sdktypes.InvalidStatus, err
+	}
+
+	st, err := i.GetConnectionStatus(ctx, id)
+	if err != nil {
+		return sdktypes.InvalidStatus, err
+	}
+
+	if err := c.Update(ctx, sdktypes.NewConnection(id).WithStatus(st)); err != nil {
+		return sdktypes.InvalidStatus, err
+	}
```

Comment:
this hides assumptions on how the db layer does updates
in case db would decide to set the clean the columns that are not set in this object (which is a valid option) - you won't get the update that you plan

I'd add an updateConnectionStatus to the db layer, which gets an ID and status, but that might be an overkill 

WDYT?

Reply:
i agree with you. but i debated if to do that and thought it's an overkill for now. we might revisit in the future.